api4/channel_test.go

```diff
@@ -1162,18 +1038,17 @@ func TestDeleteChannel(t *testing.T) {
 	pass, resp := Client.DeleteChannel(publicChannel1.Id)
 	CheckNoError(t, resp)
 
-	if !pass {
-		t.Fatal("should have passed")
-	}
+	require.True(t, pass, "should have passed")
 
-	if ch, err := th.App.GetChannel(publicChannel1.Id); err == nil && ch.DeleteAt == 0 {
-		t.Fatal("should have failed to get deleted channel")
-	}
+	ch, err := th.App.GetChannel(publicChannel1.Id)
+	require.Condition(t, func() bool {
+		return err != nil || ch.DeleteAt != 0
+	}, "should have failed to get deleted channel, or returned one with a populated DeleteAt.")
 
 	post1 := &model.Post{ChannelId: publicChannel1.Id, Message: "a" + GenerateTestId() + "a"}
-	if _, err := Client.CreatePost(post1); err == nil {
-		t.Fatal("should have failed to post to deleted channel")
-	}
+	_, resp = Client.CreatePost(post1)
+	require.NotNil(t, resp, "expected app error posting to deleted channel")
+	require.Error(t, resp.Error, "expected error in response when posting to deleted channel")
```

Comment:
It looks like the original code, which names the return `err`, is a bit misleading.  Looking at the implementation of `CreatePost`, it will always return an object of type `Response` (not an error), built with either `BuildResponse()` or a `BuildErrorResponse()`.  The response from the former has a nil `.Error`, the latter has it populated. Therefore, we would only really need the second assertion `require.Error`, because the response will always be non-nil.  To protect against the test being brittle, however, I was explicit and added both.  Let me know how you'd like me to proceed!